src/services/user.rs

```diff
@@ -96,10 +176,41 @@ pub fn validate_and_generate_token(
     jwt_secret: &str,
 ) -> Result<String> {
     let real_login_code = user::get_login_code(conn, &email)?;
+    let id = user::get_id_by_email(conn, &email)?;
 
     if login_code == real_login_code {
-        Ok(jwt::generate_token(&email, jwt_secret)?)
+        Ok(jwt::generate_token(
+            Utc::now().naive_utc(),
+            &id,
+            jwt_secret,
+        )?)
     } else {
         Err(UserServiceError::LoginCodeNotMatching)
     }
 }
+
+pub fn create_integration(
+    conn: &database::DbPool,
+    token: &str,
+    jwt_secret: &str,
+    name: String,
+    time: NaiveDateTime,
+) -> Result<UserIntegration> {
+    let id = jwt::verify_token(Utc::now().naive_utc(), token, jwt_secret)?;
```

Comment:
Can't you verify the token on the "upper" layer, as middleware? This is kinda repetitive here.

Reply:
Fixed.